File: src/main/java/io/github/intellij/dlanguage/psi/DlangItemPresentation.java
package io.github.intellij.dlanguage.psi;

import com.intellij.navigation.ItemPresentation;
import com.intellij.psi.PsiFile;
import io.github.intellij.dlanguage.DLanguage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

/**
 * @author Samael Bate (singingbush)
 * created on 20/10/18
 */
public abstract class DlangItemPresentation implements ItemPresentation {

    private final PsiFile psiFile;

    protected DlangItemPresentation(final PsiFile psiFile) {
        this.psiFile = psiFile;
    }

    @Nullable
    @Override
    public String getLocationString() {
        return psiFile instanceof DlangFile ? ((DlangFile) psiFile).getModuleOrFileName() : null;
    }

    @NotNull
    @Override
    public Icon getIcon(boolean unused) {
        return DLanguage.Icons.FILE;
    }
}

File: dlang/psi-api/src/main/kotlin/io/github/intellij/dlanguage/psi/DlangItemPresentation.kt
package io.github.intellij.dlanguage.psi

import com.intellij.navigation.ItemPresentation
import com.intellij.psi.PsiFile
import io.github.intellij.dlanguage.DLanguage
import javax.swing.Icon

/**
 * @author Samael Bate (singingbush)
 * created on 20/10/18
 */
abstract class DlangItemPresentation protected constructor(private val psiFile: PsiFile) : ItemPresentation {
    override fun getLocationString(): String? {
        return if (psiFile is DlangPsiFile) psiFile.getModuleName() else null
    }

    override fun getIcon(unused: Boolean): Icon {
        return DLanguage.Icons.FILE
    }
}
